### ISSUE TITLE
Constructor for UserTextIdentificationFrame not working has expected

### ISSUE THREAD

**Issue by jameswyper** [robinst/taglib-ruby#107]
I'm trying to create id3v2 tags with UserTextIdentificationFrames.  The documentation says that the constructor for this frame type is inherited from TextIdentificationFrame.  However,

`userframe = TagLib::ID3v2::UserTextIdentificationFrame.new("UserFrame",TagLib::String::UTF8)`

results in

```
Uncaught exception: Wrong arguments for overloaded method 'UserTextIdentificationFrame.new'.
Possible C/C++ prototypes are:
    UserTextIdentificationFrame.new(TagLib::String::Type encoding)
    UserTextIdentificationFrame.new()
    UserTextIdentificationFrame.new(TagLib::ByteVector const &data)
    UserTextIdentificationFrame.new(TagLib::String const &description, TagLib::StringList const &values, TagLib::String::Type encoding)
    UserTextIdentificationFrame.new(TagLib::String const &description, TagLib::StringList const &values)
```

From the error message I'd expect a constructor of the form

`userframe = TagLib::ID3v2::UserTextIdentificationFrame.new("UserFrame",["Some User Text"],TagLib::String::UTF8)`

to work, but that gives the same error.

```
userframe = TagLib::ID3v2::UserTextIdentificationFrame.new(TagLib::String::UTF8)
userframe.description = "UserFrame"
userframe.text = "Some User Text"
```

appears to work.

Glancing at the taglib source code I think that the constructor for UserTextIdentificationFrame genuinely is different to TextIdentificationFrame (so I believe the documentation isn't correct) but I don't understand why at least one of the correct forms reported by the error message doesn't work.  I'm not ruling out my own ignorance, so I'd appreciate any guidance.  Thanks.

**Comment by jacobvosmaer** [comment]
It appears SWIG uses some heuristics to decide which C++ constructor to call. The best advice I can give is to try and understand how those heuristics work and adjust your Ruby call accordingly.

https://github.com/robinst/taglib-ruby/blob/dc7858f1d359f7697fc1758251f63e640d13f8ab/ext/taglib_id3v2/taglib_id3v2_wrap.cxx#L9190-L9263

But that error message already gives you an overview of the options.

As you can see in the link, SWIG looks at the number of Ruby arguments and their types in a fixed order and if anything matches, it calls the corresponding C++ constructor. If nothing matches you get the error.

First it looks if there is exactly one argument and if it can be coerced to an integer. If so it uses the first mentioned constructor. Then it checks if there are zero arguments, and if so it uses the second constructor. Etc.

**Comment by robinst** [comment]
Hey @jameswyper, I revisited this and found a fix, see here: https://github.com/robinst/taglib-ruby/pull/120

**Comment by robinst** [comment]
Released the fix with version 1.1.2 now.
